docs/src/main/asciidoc/spanner.adoc

```diff
@@ -25,51 +27,180 @@ dependencies {
 }
 ----
 
+This setup takes care of bringing in the latest compatible version of Cloud Java Spanner libraries as well.
+
+== Spring Data Spanner
+
+
+=== Configuration
+
+To setup Spring Data Spanner, you have to configure connection to Cloud Spanner and Repositories (if you want to use them).
+
+==== Spanner settings
+
+You will need to use link:../spring-cloud-gcp-starters/spring-cloud-gcp-starter-data-spanner[Spring Boot Starter for Spring Data Spanner] to configure Google Cloud Spanner in your Spring application. It contains all the necessary setup that makes it easy to authenticate with your Google Cloud project.
+The following configuration options are available:
+
+|===
+| Name | Description | Optional | Default value
+| `spring.cloud.gcp.spanner.projectId` | the GCP project ID to use | Yes | provided by Spring Cloud GCP
+| `spring.cloud.gcp.spanner.instanceId` | Spanner instance to use | No |
+| `spring.cloud.gcp.spanner.database` |
+Spanner database to use | No |
+|===
+
+==== Repository settings
+
+Spring Data Repositories (see below for functionality) can be configured via the `@EnableSpannerRepositories` annotation on your main configuration file (typically the `@SpringBootApplication` file.
+
+==== Available beans
+
+With our Spring Boot autoconfiguration your application is able to inject an instance of `DatabaseClient` and `SpannerOperations`, and repositories are enabled, all the user defined repositories extending `SpannerRepository` are exposed as beans as well.
```

Comment:
I'd rename this section to `Autoconfiguration` and change the text a bit to mention that a `SpannerTemplate` (not `SpannerOperations`, to follow Pub/Sub and, in general, Spring preferred terminology) object is made available in the Spring application context.

If I understand correctly, `DatabaseClient` is also made available as a kind of utility, if you want lower level access, so it should be conveyed in that away, after mentioning `SpannerTemplate`.